pkg/config/v1alpha5/schema.json

```diff
@@ -114,6 +114,40 @@
         "additionalProperties": false
       }
     },
+    "files": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "required": ["destination"],
+        "properties": {
+          "path": {
+            "type": "string"
+          },
+          "content": {
+            "type": "string"
+          },
+          "destination": {
+            "type": "string"
+          },
+          "description": {
+            "type": "string"
+          }
+        },
+        "oneOf": [
```

Comment:
@iwilltry42 The fields `path` (for file path relative to k3d config) and `content` (for file content embedded in the k3d config) are mutually exclusive. So I'm checking that in the schema, which returns a vague message at the same time `gojsonschema` [doesn't support a custom message](https://github.com/xeipuuv/gojsonschema/issues/371) for `oneOf`. Do you think it's a good idea to do it in the implementation?

Reply:
If the error message is that bad, I'd go for struct validation on the implementation, yep

Reply:
It's so bad that I'm barely able to know the issue when I read the error even though I know exactly where it comes from 😂

I will revisit it again once we decide if we will go with that path or just use "source".